app/src/main/java/com/compscieddy/meetinthemiddle/InternetErrorFragment.java

```diff
@@ -49,16 +49,16 @@ public void onResume() {
     getDialog().getWindow().setAttributes((android.view.WindowManager.LayoutParams) params);
 
     intentFilter = new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE");
-    netWorkChangeReceiver = new NetworkChangeReceiver();
-    getActivity().registerReceiver(netWorkChangeReceiver, intentFilter);
+    networkChangeReceiver = new NetworkChangeReceiver();
+    getActivity().registerReceiver(networkChangeReceiver, intentFilter);
 
     super.onResume();
   }
 
   @Override
   public void onDismiss(DialogInterface dialog) {
     super.onDismiss(dialog);
-    final Activity activity = getActivity();
+    Activity activity = getActivity();
     if (activity instanceof DialogInterface.OnDismissListener){
```

Comment:
Notice I didn't tell you to do an "activity == null" check because instanceof actually handles that for you